orchagent/port_flr.lua

```diff
@@ -0,0 +1,460 @@
+    local F = 0     -- ## sigma(xy)
+    local G = 0     -- ## sigma(y^2)
+
+    for i = 1, data_length do
+        if mask[i] == 1 then
+            B = B + mask[i]
+            C = C + (i)
+            D = D + (log_values_cw[i])
+            E = E + ((i) * (i))
+            F = F + ((i) * (log_values_cw[i]))
+            G = G + ((log_values_cw[i]) * (log_values_cw[i]))
+        end
+    end
+
+    -- Slope and Intercept
+    local slope = (B*F - C*D)/(B*E - C*C)
+    local intercept = (D - slope*C) / B
+
+    -- R^2 (measure of accuracy)
+    local numerator = (B * F - C * D)
+    local denominator = math.sqrt((B * E - C*C) * (B * G - D*D))
+    local r_squared = (numerator / denominator) * (numerator / denominator)
+
+    return slope, intercept, r_squared
+end
+
+-- Compute the predicted Frame Loss Ratio (FLR) from linear regression parameters
+-- Uses the fitted slope and intercept to extrapolate CER for uncorrectable symbol errors
+-- (window 16-20) and converts to FLR using IEEE FEC formula with interleaving factor.
+-- Parameters:
+--   slope: Fitted slope from log-linear regression
+--   intercept: Fitted intercept from log-linear regression
+--   sum_window: Array of [start_index, end_index] for uncorrectable error window (16-20)
+--   x_interleaving: FEC interleaving factor (1=no interleaving, 2=400G, 4=800G+)
+--   mfc: MAC frames per codeword (8 for RS-544 FEC)
+-- Returns:
+--   cer: Predicted Codeword Error Ratio for the window
+--   flr: Predicted Frame Loss Ratio using IEEE formula
+local function extrapolate_flr_from_regression(slope, intercept, sum_window, x_interleaving, mfc)
+    -- Transform logarithmic regression line back to linear scale to get predicted CER
+    local function line_function(x)
+        return 10 ^ (intercept + (slope*x))
+    end
+
+    -- # Sum predicted corrected errors in the given window
+    local cer = 0
+    local flr = 0
+    for x = sum_window[1], sum_window[2]+1 do
+        cer = cer + line_function(x)
+    end
+
+    -- # IEEE FLR formula
+    flr = cer * (1 + x_interleaving * mfc) / mfc
+
+    return cer, flr
+end
+
+-- Main function to calculate predicted FLR using linear regression on codeword error distribution
+-- Steps: Get error deltas -> Remove S0 -> Check sufficient data -> Perform regression -> Extrapolate FLR
+-- Parameters:
+--   port: Port identifier
+-- Returns:
+--   flr: Predicted Frame Loss Ratio (0 if insufficient data for prediction)
+local function compute_predicted_flr(port)
+    local bins = get_fec_cws_delta(port)
+
+    local total_cws = get_total_cws(bins)
+    logit("SUM : " .. total_cws)
+    if total_cws == 0 then
+        logit("Total corrected codewords is zero, cannot compute slope and intercept.")
+        return 0
+    end
+
+    -- Trim out _S0 from the data
+    table.remove(bins, 1)
+
+    local significant_bins = count_significant_bins(bins)
+    logit("Significant Bins : " .. significant_bins)
+    if significant_bins < 2 then
```

Comment:
@apoorv-arista can you make significant as enum value defined at the top for easy change in future?